tests/test_amgx_direct_solver.cu

```diff
@@ -176,7 +176,8 @@ BOOST_AUTO_TEST_CASE(amgx_2_procs)
         cusparseSetMatIndexBase(matrix_dev->descr, CUSPARSE_INDEX_BASE_ZERO);
     mfmg::ASSERT_CUSPARSE(cusparse_error_code);
     auto rhs_dev = mfmg::copy_from_host(rhs);
-    auto solution_dev = mfmg::copy_from_host(sol_ref);
+    auto solution_dev = mfmg::copy_from_host(rhs);
+    solution_dev = 0.;
```

Comment:
Was it wrong to copy `sol_ref`?  They seem to be both built with the same index range.  Was the issue initialization to zero?  I thought deal.II vector constructors where usually doing the initialization to zero, so copying from `rhs` would already have the zeroes.

Reply:
No, this wasn't necessarily wrong. The reasoning is just to not provide the solution as initial guess to the solver (unless we want to test for this corner case).
I first tested with `solution_dev` initialized by `rhs`, but then decided to go for zero-initialization to match the serial test case.
We could do
```
dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::CUDA>
solution_dev(sol_ref.get_partitioner());
```
instead if you prefer that.

Reply:
Sure